src/v1-promise.test.ts

```diff
@@ -0,0 +1,227 @@
+    );
+  });
+});
+
+describe('Lookup APIs', () => {
+  let token: string;
+
+  beforeEach(async () => {
+    token = `v1-lookupsubject-${Math.floor(Math.random() * 1000)}`;
+    const { promises: client } = NewClient(
+      token,
+      'localhost:50051',
+      ClientSecurity.INSECURE_LOCALHOST_ALLOWED
+    );
+
+    const request = WriteSchemaRequest.create({
+      schema: `definition test/user {}
+
+      definition test/document {
+        relation viewer: test/user
+        permission view = viewer
+      }
+      `,
+    });
+
+    await client.writeSchema(request)
+
+    const resource = ObjectReference.create({
+      objectType: 'test/document',
+      objectId: 'somedocument',
+    });
+
+    const writeRequest = WriteRelationshipsRequest.create({
+      updates: [
+        RelationshipUpdate.create({
+          relationship: Relationship.create({
+            resource: resource,
+            relation: 'viewer',
+            subject: SubjectReference.create({
+              object: ObjectReference.create({
+                objectType: 'test/user',
+                objectId: 'someuser',
+              }),
+            }),
+          }),
+          operation: RelationshipUpdate_Operation.CREATE,
+        }),
+        RelationshipUpdate.create({
+          relationship: Relationship.create({
+            resource: resource,
+            relation: 'viewer',
+            subject: SubjectReference.create({
+              object: ObjectReference.create({
+                objectType: 'test/user',
+                objectId: 'someuser2',
+              }),
+            }),
+          }),
+          operation: RelationshipUpdate_Operation.CREATE,
+        }),
+      ],
+    });
+
+    return client.writeRelationships(writeRequest)
+  });
+
+  it('can lookup subjects', async () => {
+    const { promises: client } = NewClient(
+      token,
+      'localhost:50051',
+      ClientSecurity.INSECURE_LOCALHOST_ALLOWED
+    );
+
+    const request = LookupSubjectsRequest.create({
+      resource: ObjectReference.create({
+        objectType: 'test/document',
+        objectId: 'somedocument',
+      }),
+      permission: 'view',
+      subjectObjectType: 'test/user',
+    });
+
+    const result = await client.lookupSubjects(request)
+    expect(['someuser', 'someuser2']).toContain(result[0].subjectObjectId);
+  });
+
+  it('can lookup resources', async () => {
+    const { promises: client } = NewClient(
+      token,
+      'localhost:50051',
+      ClientSecurity.INSECURE_LOCALHOST_ALLOWED
+    );
+
+    const request = LookupResourcesRequest.create({
+      subject: SubjectReference.create({
+        object: ObjectReference.create({
+          objectType: 'test/user',
+          objectId: 'someuser',
+        }),
+      }),
+      permission: 'view',
+      resourceObjectType: 'test/document',
+    });
+
+    const resStream = await client.lookupResources(request)
+    expect(resStream[0].resourceObjectId).toEqual('somedocument');
```

Comment:
Getting this error:
`TypeError: Cannot read properties of undefined (reading 'resourceObjectId')`

Reply:
This was due to consistency not being set on the original lookup tests - fixed!